src/core/Akka.API.Tests/CoreAPISpec.ApproveClusterSharding.approved.txt

```diff
@@ -156,70 +179,42 @@ namespace Akka.Cluster.Sharding
-        protected System.Collections.Immutable.IImmutableSet<Akka.Cluster.Member> MembersByAge;
-        protected System.Collections.Immutable.IImmutableDictionary<string, Akka.Actor.IActorRef> RegionByShard;
-        protected System.Collections.Immutable.IImmutableDictionary<Akka.Actor.IActorRef, System.Collections.Immutable.IImmutableSet<string>> Regions;
-        public readonly Akka.Cluster.Sharding.ClusterShardingSettings Settings;
-        protected System.Collections.Immutable.IImmutableDictionary<string, System.Collections.Immutable.IImmutableList<System.Collections.Generic.KeyValuePair<object, Akka.Actor.IActorRef>>> ShardBuffers;
-        protected System.Collections.Immutable.IImmutableDictionary<string, Akka.Actor.IActorRef> Shards;
-        protected System.Collections.Immutable.IImmutableDictionary<Akka.Actor.IActorRef, string> ShardsByRef;
-        protected System.Collections.Immutable.IImmutableSet<string> StartingShards;
-        public readonly string TypeName;
-        public ShardRegion(string typeName, System.Func<string, Akka.Actor.Props> entityProps, Akka.Cluster.Sharding.ClusterShardingSettings settings, string coordinatorPath, Akka.Cluster.Sharding.ExtractEntityId extractEntityId, Akka.Cluster.Sharding.ExtractShardId extractShardId, object handOffStopMessage, Akka.Actor.IActorRef replicator, int majorityMinCap) { }
-        public bool GracefulShutdownInProgress { get; }
-        public Akka.Event.ILoggingAdapter Log { get; }
-        protected object RegistrationMessage { get; }
+        public ShardRegion(string typeName, System.Func<string, Akka.Actor.Props> entityProps, Akka.Cluster.Sharding.ClusterShardingSettings settings, string coordinatorPath, Akka.Cluster.Sharding.ExtractEntityId extractEntityId, Akka.Cluster.Sharding.ExtractShardId extractShardId, object handOffStopMessage, Akka.Cluster.Sharding.Internal.IRememberEntitiesProvider rememberEntitiesProvider) { }
         public Akka.Actor.ITimerScheduler Timers { get; set; }
-        public int TotalBufferSize { get; }
-        protected bool MatchingRole(Akka.Cluster.Member member) { }
         protected override void PostStop() { }
         protected override void PreStart() { }
         protected override bool Receive(object message) { }
-        public sealed class StartEntity : Akka.Cluster.Sharding.IClusterShardingSerializable
+        public sealed class StartEntity : Akka.Cluster.Sharding.IClusterShardingSerializable, System.IEquatable<Akka.Cluster.Sharding.ShardRegion.StartEntity>
         {
             public readonly string EntityId;
             public StartEntity(string entityId) { }
             public override bool Equals(object obj) { }
+            public bool Equals(Akka.Cluster.Sharding.ShardRegion.StartEntity other) { }
             public override int GetHashCode() { }
-        }
-        public sealed class StartEntityAck : Akka.Cluster.Sharding.IClusterShardingSerializable, Akka.Event.IDeadLetterSuppression
```

Comment:
Any reason why we remove this class?

Reply:
it's now internal, should not be used directly